### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by camirmas**
**Do not merge**. This jam needs tests

Additions to original branch, `scheduler`:
- Uses a more generic `{module, function, arguments}` tuple, which is converted to a binary and saved in Mnesia, then converted back and called with `apply` when needed in `Scheduler.Consumer`. This is intended to allow more flexible use by other applications.
- Gives `Job` a `status`, which starts as "pending" and changes to "failed" if the job does not succeed. Failed jobs stick around so they can be introspected, though how to deal with these is a topic for discussion

### CODE REVIEW

File: apps/scheduler/lib/job/pipeline/consumer.ex [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Enum.each(jobs, fn(job) ->
+      execute_job(job)
+    end)
+
+    {:noreply, [], state}
+  end
+
+  defp execute_job(job) do
+    {m, f, a} = :erlang.binary_to_term(job.params)
+    with :ok <- Kernel.apply(m, f, a),
```
Comment (msmykowski) [inline apps/scheduler/lib/job/pipeline/consumer.ex]: does the scheduler have a sense of what these functions are, or do we need to add the applications that define these functions as dependencies of the scheduler?

File: apps/scheduler/lib/job/job.ex [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  @doc """
+  Creates a new Job. It takes a name, a time to execute (in erlang system
+  time, seconds) and a tuple of {module, fun, args} that will be executed
+  when the job is run.
+  """
+  def create(name, execute_at, mfa) do
+    Amnesia.transaction do
+      %Job{
+        name: name,
```
Comment (msmykowski) [inline apps/scheduler/lib/job/job.ex]: if we use the params, we can get rid of the name attribute

File: apps/scheduler/lib/job/job.ex [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        execute_at: execute_at,
+        params: :erlang.term_to_binary(mfa),
+        status: "pending",
```
Comment (msmykowski) [inline apps/scheduler/lib/job/job.ex]: great idea. yea we will need to decide how to handle these

File: apps/scheduler/lib/job/pipeline/consumer.ex (hunk 1 above)
Comment (camirmas) [reply apps/scheduler/lib/job/pipeline/consumer.ex]: No we should be good here, though we should add `scheduler` as a dep in whatever app needs to implement jobs

Comment (camirmas) [reply apps/scheduler/lib/job/pipeline/consumer.ex]: I tried with an example job that I put in API and it was working fine, I can check it in if helpful

File: apps/scheduler/lib/job/job.ex (hunk 3 above)
Comment (camirmas) [reply apps/scheduler/lib/job/job.ex]: A quick but still reasonable option could be to let the client determine when to delete them, providing an API to do so and leaving it at that

**Pull request merged**